nipoppy/zenodo_api.py

```diff
@@ -52,9 +47,18 @@ def set_authorization(self, access_token: str):
         """Set the headers for the ZenodoAPI instance."""
         self.headers.update({"Authorization": f"Bearer {access_token}"})
 
-    def set_logger(self, logger: logging.Logger):
+    @property
+    def logger(self) -> logging.Logger:
+        """Get the logger for the ZenodoAPI instance."""
+        return self._logger
+
+    @logger.setter
+    def logger(self, logger: logging.Logger):
```

Comment:
Would this be more accurate for the type annotation?

```suggestion
    def logger(self, logger: logging.Logger | None):
```